### PR TITLE
Adding command layer 🚀. Outlining layers scopes.

### PR DESCRIPTION
**Pull request by avgvsto**
First "usable" version of the command.

1) Added the **command** layer. Using [click](https://click.palletsprojects.com/) to manage the command line interface (arguments, validations, echoing, ...).

2) Refactored the **client** layer, it's limited to dispatch the request (backend `do_request`) and beautify the response (Exception/ actual response), this last is parsing to text and formatting minimally.

3) Added a `do_request` generic function to the **backend** layer. This aims to maintain all the HTTP related functions in this layer. E.i: in this commit it is charge of validating the allowed methods (GET only for now).

### CODE REVIEW

File: backend_http/do_request.py [hunk 1]
```diff
@@ -7,6 +7,19 @@ class InvalidUrl(Exception):
     pass
 
 
+class NotImplementedHttpVerb(Exception):
+    pass
+
+
+def do_request(http_verb, url):
+
+    if http_verb in ('get', 'GET'):
```
Comment (eb-fmezzabotta) [inline backend_http/do_request.py]: `http_verb.lowercase() == 'get'` para aceptar cualquier combinación de mayúsculas/minúsculas?

File: backend_http/do_request.py [hunk 2]
```diff
@@ -7,6 +7,19 @@ class InvalidUrl(Exception):
     pass
 
 
+class NotImplementedHttpVerb(Exception):
```
Comment (eb-fmezzabotta) [inline backend_http/do_request.py]: nit: HttpVerbNotImplemented

File: tests/client/test_client.py [hunk 3]
```diff
@@ -1,21 +1,42 @@
...
-        mock_get.assert_called_once_with('')
+        mock_do_request.assert_called_once_with('get', '')
         assert response == 'No request: the provided url is invalid.'
 
-    def test_it_handles_backend_responses(self, mock_get):
+    @mock.patch('client.client_request.do_request')
+    def test_it_maps_not_implemented_http_verb_to_text(self, mock_do_request):
+
+        mock_do_request.side_effect = NotImplementedHttpVerb
+        response = client_request('post', 'http://lemon.com/api/resource')
```
Comment (eb-fmezzabotta) [inline tests/client/test_client.py]: urls con trailing slash incluso el los tests? :p

File: client/client_request.py [hunk 4]
```diff
@@ -1,14 +1,25 @@
-from backend_http.get import get, InvalidUrl
+from backend_http.do_request import (
+    InvalidUrl,
+    NotImplementedHttpVerb,
+    do_request,
```
Comment (eb-fmezzabotta) [inline client/client_request.py:4]: A este método lo llamaría `make_request` o `perform_request`.

File: client/client_request.py [hunk 5]
```diff
@@ -1,14 +1,25 @@
-from backend_http.get import get, InvalidUrl
+from backend_http.do_request import (
```
Comment (eb-fmezzabotta) [inline client/client_request.py:1]: Este módulo lo generalizaría como `requests` en vez de `do_request`.

**Review by eb-fmezzabotta — APPROVED** [review]

**Comment by avgvsto** [comment]
Applying some Fernando's comments. Let's leave the module/ function renaming for a further commit.

**Pull request merged**